code/modules/atmospherics/portable/portable_pressurizer.dm

```diff
@@ -0,0 +1,412 @@
+				switch(target_material.material.name)
+					if("molitz")
+						GM.oxygen += 1500 * progress / 100
+					if("viscerite")
+						GM.nitrogen += 1500 * progress / 100
+					if("char")
+						GM.carbon_dioxide += 500 * progress / 100
+					if("plasmastone")
+						GM.toxins += 500 * progress / 100
+			else
+				playsound(src.loc, 'sound/machines/buzz-two.ogg', 20)
+				process_materials = PROCESS_PAUSED
+				target_material?.set_loc(src.loc)
+				target_material = null
+				return
+			src.air_contents.merge(GM)
+			material_progress = clamp(material_progress + progress, 0, 100)
+			use_power(src.powconsumption)
+		else
+			qdel(target_material)
+			target_material = null
+
+	emag_act(var/mob/user, var/obj/item/card/emag/E)
+		if (!src.emagged)
+			if (user)
+				user.show_text("You short out the material processor on [src].", "red")
+			src.audible_message("<span class='combat'><B>[src] buzzes oddly!</B></span>")
+			playsound(src.loc, "sparks", 50, 1, -1)
+			whitelist += blacklist
+			src.emagged = TRUE
+			return 1
+		return 0
+
+	demag(var/mob/user)
+		if (!src.emagged)
+			return 0
+		if (user)
+			user.show_text("You repair [src]'s material processor.", "blue")
+		src.emagged = FALSE
+		src.process_rate = min(src.process_rate, 3)
+		whitelist = initial(whitelist)
+		return 1
+
+	// attack by item places it in to disposal
+	attackby(obj/item/I, mob/user)
+		if(status & BROKEN)
+			return
+		if(iswrenchingtool(I) || istype(I,/obj/item/device/analyzer/atmospheric) || istype(I,/obj/item/card/emag))
+			..()
+			return
+		if(istype(I,/obj/item/electronics/scanner) || istype(I,/obj/item/deconstructor) || (istype(I,/obj/item/device/pda2)))
+			user.visible_message("<span class='alert'><B>[user] hits [src] with [I]!</B></span>")
+			return
+		if (istype(I,/obj/item/satchel/) && I.contents.len)
+			var/obj/item/satchel/S = I
+			for(var/obj/item/O in S.contents) O.set_loc(src)
+			S.satchel_updateicon()
+			user.visible_message("<b>[user.name]</b> dumps out [S] into [src].")
+			return
+		if (istype(I,/obj/item/storage/) && I.contents.len)
+			var/obj/item/storage/S = I
+			for(var/obj/item/O in S)
+				O.set_loc(src)
+				S.hud.remove_object(O)
+			user.visible_message("<b>[user.name]</b> dumps out [S] into [src].")
+			return
+
+		var/obj/item/grab/G = I
+		if(istype(G))	// handle grabbed mob
+			if(ismob(G.affecting))
+				boutput(user, "<span class='alert'>That won't fit!</span>")
+				return
+		else
+			if(!user.drop_item())
+				return
+			else if(I.w_class > W_CLASS_NORMAL)
+				boutput(user, "<span class='alert'>That won't fit!</span>")
+				return
+			I.set_loc(src)
+			user.visible_message("[user.name] places \the [I] into \the [src].",\
+			"You place \the [I] into \the [src].")
+			actions.interrupt(user, INTERRUPT_ACT)
+
+	// eject the contents of the unit
+	proc/eject()
+		for(var/atom/movable/AM in src)
+			AM.set_loc(src.loc)
+
+	proc/arm_blast()
+		if(src.blast_armed)
+			src.blast_armed = FALSE
+		else
+			var/blast_key = rand()
```

Comment:
What is `blast_key`? 🤔

Reply:
Unique ID to ensure spawned process matches currently active detonation to improve time resolution. To avoid A: Activate B: Deactivate C: Activate, A detonation was spawned so it doesn't know it was deactivated. Essentially a transaction identifier.